test/suite/Annotations.test.php

```diff
@@ -585,12 +592,72 @@ protected function testGetAnnotationsFromTrait()
             return;
         }
 
-        eval('trait TestTrait { }');
-        $this->setExpectedException(
-            self::ANNOTATION_EXCEPTION,
-            "Reading annotations from interface/trait 'TestTrait' is not supported"
-        );
-        Annotations::ofClass('TestTrait', '@note');
+        $annotations = Annotations::ofClass('SimpleTrait', '@note');
+
+        $this->check(count($annotations) === 1, 'SimpleTrait has one note annotation.');
+    }
+
+    protected function testCanGetMethodAnnotationsIncludedFromTrait()
```

Comment:
The scenario, where traits declare methods on different deep levels isn't covered by tests:

```php
trait Level1Trait {
    use Level2Trait;

    public function level1Method();
}

trait Level2Trait {
    public function level2Method();
}

class MultiLevelClass {
    use Level1Trait;
}
```

In this case the `MultiLevelClass` should have methods from both traits including their annotations.

Looking at `AnnotationParser` code this case might already work, but without tests to prove it I'm not 100% certain.